### PR TITLE
envs/glulx: Don't assume that closed tags will have been opened in the same action

### LINKED ISSUE #168: Error when deciding not to quit a game with `tw-play`
I understand this is probably not the most important bug to fix, but I found it while manually inspecting some generated games.

To reproduce, first create a game with:

```
tw-make custom --world-size 5 --nb-objects 10 --quest-length 5 --seed 1234 --output tw_games/custom_game.ulx
```

Then begin to play the game with:

```
tw-play tw_games/custom_game.ulx
```

Immediately try and quit the game with "q" or "quit", when it asks "Are you sure you want to quit?" type either "n" or "no". This produces the error:

```
Traceback (most recent call last):
  File "/home/ben/miniconda3/envs/pytorch10/bin/tw-play", line 82, in <module>
    main()
  File "/home/ben/miniconda3/envs/pytorch10/bin/tw-play", line 69, in main
    game_state, reward, done = env.step(command)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/envs/glulx/git_glulx_ml.py", line 541, in step
    self.game_state = self.game_state.update(command, output)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/envs/glulx/git_glulx_ml.py", line 256, in update
    i7_events, game_state._feedback = _detect_i7_events_debug_tags(output)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/envs/glulx/git_glulx_ml.py", line 139, in _detect_i7_events_debug_tags
    open_tags.remove(tag_name)
ValueError: list.remove(x): x not in list
```

### PR DESCRIPTION
**Pull request by tavianator**
This fixes #168.

### CODE REVIEW

**Comment by MarcCote** [comment]
Actually, we might want to detect Inform7 tags that span multiple turns :-/. For instance, issuing the `restart` command and answering yes, completely break the state tracking in GitGlulxML.

**Comment by MarcCote** [comment]
@tavianator if I remember correctly, we wanted to make a small change (i.e. only considering the ending Inform7 event tag) to this PR before merging it, right?

**Comment by tavianator** [comment]
@MarcCote Right, done

File: textworld/envs/glulx/git_glulx_ml.py [hunk 1]
```diff
@@ -125,29 +125,21 @@ def _detect_i7_events_debug_tags(text: str) -> Tuple[List[str], str]:
         in the text, and a cleaned text without Inform 7 debug infos.
     """
     matches = []
-    open_tags = []
     for match in re.findall(r"\[[^]]+\]\n?", text):
         text = text.replace(match, "")  # Remove i7 debug tags.
         tag_name = match.strip()[1:-1]  # Strip starting '[' and trailing ']'.
 
         if " - failed" in tag_name:
```
Comment (MarcCote) [inline textworld/envs/glulx/git_glulx_ml.py]: @tavianator it seems now we only need to check for successful events. This `if` statement doesn't do anything now.

Comment (tavianator) [reply textworld/envs/glulx/git_glulx_ml.py]: Good point, I removed this `if`

**Pull request merged**